src/formElementPlugins/listBuilder/src/ApplicationView.tsx

```diff
@@ -0,0 +1,339 @@
+import React, { useEffect, useState } from 'react'
+import {
+  Button,
+  Modal,
+  Form,
+  Segment,
+  Table,
+  TableHeaderCell,
+  TableCell,
+  Card,
+  Icon,
+  Label,
+} from 'semantic-ui-react'
+import { ApplicationViewProps } from '../../types'
+import { ResponseFull } from '../../../utils/types'
+import { TemplateElement, TemplateElementCategory } from '../../../utils/generated/graphql'
+import ApplicationViewWrapper from '../../ApplicationViewWrapper'
+import strings from '../constants'
+
+enum DisplayType {
+  CARDS = 'cards',
+  TABLE = 'table',
+}
+
+interface InputResponseField {
+  isValid?: boolean
+  value: ResponseFull
+}
+
+type ListItem = { [code: string]: InputResponseField }
+
+const ApplicationView: React.FC<ApplicationViewProps> = ({
+  element,
+  parameters,
+  onSave,
+  onUpdate,
+  Markdown,
+  initialValue,
+  currentResponse,
+  applicationData,
+  allResponses,
+}) => {
+  const { isEditable } = element
+  const {
+    label,
+    description,
+    createModalButtonText = strings.BUTTON_LAUNCH_MODAL,
+    modalText,
+    addButtonText = strings.BUTTON_ADD,
+    updateButtonText = strings.BUTTON_UPDATE,
+    deleteItemText = strings.BUTTON_DELETE,
+    inputFields,
+    displayFormat = getDefaultDisplayFormat(inputFields),
+    displayType = DisplayType.CARDS,
+  } = parameters
+
+  const [currentInputResponses, setCurrentInputResponses] = useState<ListItem>(
+    resetCurrentResponses(inputFields)
+  )
+
+  const [listItems, setListItems] = useState<ListItem[]>(initialValue?.list ?? [])
+  const [selectedListItem, setSelectedListItem] = useState<number | null>(null)
+  const [open, setOpen] = useState(false)
+  const [inputError, setInputError] = useState(false)
+
+  useEffect(() => {
+    onSave({
+      text: listItems.length > 0 ? createTextString(listItems, inputFields) : undefined,
+      list: listItems,
+    })
+  }, [listItems])
+
+  // This is sent to ApplicationViewWrapper and runs instead of the default responseMutation
+  const innerElementUpdate =
+    (code: string) =>
+    async ({ variables: response }: { variables: InputResponseField }) => {
+      const newResponses = { ...currentInputResponses, [code]: response }
+      setCurrentInputResponses(newResponses)
+      if (!anyErrorItems(newResponses, inputFields)) setInputError(false)
+    }
+
+  const updateList = async () => {
+    if (anyErrorItems(currentInputResponses, inputFields)) {
+      setInputError(true)
+      return
+    }
+    // Add item
+    if (selectedListItem === null) {
+      setListItems([...listItems, currentInputResponses])
+    } else {
+      // Or update existing item
+      const newList = [...listItems]
+      newList[selectedListItem] = currentInputResponses
+      setListItems(newList)
+    }
+    setCurrentInputResponses(resetCurrentResponses(inputFields))
+    setOpen(false)
+    setSelectedListItem(null)
+  }
+
+  const editItem = async (index: number) => {
+    if (!isEditable) return
```

Comment:
hmm, just for consistency sake, would be good to either remove or also add to deleteItem

Reply:
It's actually only needed on `editItem`, as for the delete method, the UI elements that trigger it are either hidden or not reachable.

You still think it should have it just for consistency?

Reply:
How about like this? (override the methods rather than logic in the method)

Reply:
that commit looks good, thanks !